Assignment6/Array.cs

```diff
@@ -1,10 +1,91 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 
+#pragma warning disable CS8653
+#pragma warning disable CS8602
+#pragma warning disable CA2208
+#pragma warning disable CA1303
+
 namespace Assignment6
 {
-    public class Array<T>
+    public class ArrayCollection<T> : ICollection<T>
     {
-        
+        public int Count => _Data.Count;
+        public bool IsReadOnly => false;
+        public int Capacity { get; } 
+        private readonly List<T> _Data;
+        public ArrayCollection(int cap)
+        {
+            if(cap < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(cap));
+            }
+            Capacity = cap;
+            _Data = new List<T>(cap);
+        }
+
+        public ArrayCollection(ICollection<T> collection)
+        {
+            if (collection is null)
+            {
+                throw new ArgumentNullException("Collection is null", nameof(collection));

```

Comment:
```suggestion
                throw new ArgumentNullException(nameof(collection), "Collection is null");
```
Wrong order for params.